devices/Lp3943/Lp3943.cs

```diff
@@ -0,0 +1,185 @@
+
+			_controller.OpenPin(_pinReset, PinMode.Output);
+			Reset();
+		}
+
+		/// <summary>
+		/// Cleanup
+		/// </summary>
+		public void Dispose()
+		{
+			if (_i2cDevice is not null)
+			{
+				_i2cDevice?.Dispose();
+				_i2cDevice = null;
+			}
+
+			if (_pinReset >= 0)
+			{
+				_controller?.ClosePin(_pinReset);
+			}
+
+			if (_shouldDispose)
+			{
+				_controller?.Dispose();
+				_controller = null;
+			}
+		}
+
+		private void SetFrequency(DimRegister dimRegister, int frequency)
+		{
+			var toWrite = (byte)(frequency * 160 - 1);
+			var register = dimRegister switch
+			{
+				Device.Lp3943.DimRegister.Dim0 => Register.Psc0, Device.Lp3943.DimRegister.Dim1 => Register.Psc1,
+			};
+
+			WriteToRegister(register, toWrite);
+		}
+
+		private void SetDimPercentage(DimRegister dimRegister, int percentage)
+		{
+			var toWrite = (byte)((float)percentage / 100 * 256);
+			var register = dimRegister switch
+			{
+				Device.Lp3943.DimRegister.Dim0 => Register.Pwm0, Device.Lp3943.DimRegister.Dim1 => Register.Pwm1,
+			};
+
+			WriteToRegister(register, toWrite);
+		}
+
+		private void WriteToRegister(Register register, byte data)
+		{
+			var message = new byte[2];
+			message[0] = (byte)register;
+			message[1] = data;
+			_i2cDevice.Write(message);
+		}
+
+		private byte FillSelectorRegister(LedState led0, LedState led1, LedState led2, LedState led3)
+		{
+			byte result = 0x00;
+			result |= (byte)(((byte)led3 & 0x03) << 6);
+			result |= (byte)(((byte)led2 & 0x03) << 4);
+			result |= (byte)(((byte)led1 & 0x03) << 2);
+			result |= (byte)((byte)led0 & 0x03);
+			return result;
+		}
+
+		/// <summary>
+		/// Sets the dimming on a register
+		/// </summary>
+		/// <param name="register">register to dim</param>
+		/// <param name="frequency">frequency in Hz</param>
+		/// <param name="dimPercentage">percentage</param>
+		public void DimRegister(DimRegister register, int frequency, int dimPercentage)
+		{
+			if (frequency is > 160 or < 1)
+				throw new ArgumentOutOfRangeException(nameof(frequency), "The frequency has to be between 1 and 160 Hz.");
+
+			if (dimPercentage is > 100 or < 0)
+				throw new ArgumentOutOfRangeException(nameof(dimPercentage));
+
+			SetFrequency(register, frequency);
+			SetDimPercentage(register, dimPercentage);
+		}
+
+		private void Update()
+		{
+			var ls0 = FillSelectorRegister(_ledStates[0], _ledStates[1], _ledStates[2], _ledStates[3]);
+			var ls1 = FillSelectorRegister(_ledStates[4], _ledStates[5], _ledStates[6], _ledStates[7]);
+			var ls2 = FillSelectorRegister(_ledStates[8], _ledStates[9], _ledStates[10], _ledStates[11]);
+			var ls3 = FillSelectorRegister(_ledStates[12], _ledStates[13], _ledStates[14], _ledStates[15]);
+
+			WriteToRegister(Register.Ls0, ls0);
+			WriteToRegister(Register.Ls1, ls1);
+			WriteToRegister(Register.Ls2, ls2);
+			WriteToRegister(Register.Ls3, ls3);
+		}
+
+		/// <summary>
+		/// Resets the Lp3943 chip
+		/// </summary>
+		public void Reset()
```

Comment:
Does it has an interest to be exposed? is there a good case to reset it while already initialized and running?